src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OpenTelemetry.Exporter.OpenTelemetryProtocol.csproj

```diff
@@ -8,6 +8,9 @@
     <!-- this is temporary. will remove in future PR. -->
     <Nullable>disable</Nullable>
     <DefineConstants>BUILDING_INTERNAL_PERSISTENT_STORAGE;$(DefineConstants)</DefineConstants>
+    <EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>
+    <!-- SYSLIB1100;SYSLIB1101 - Configuration.Binder: can't create instance and unsupported type -->
+    <NoWarn>$(NoWarn);SYSLIB1100;SYSLIB1101</NoWarn>
```

Comment:
Oh sorry, you meant the `SYSLIB*` warnings - I don't know - @eerhardt presumably found them locally when working on his commit.

Reply:
These warnings are documented in https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1100-1118. The code that is emitting those warnings is from the Configuration Binder Source Generator. It is happening because the `IOtlpExporterOptions` public property is an interface, which can't be instantiated. So the config binder emits a warning saying "FYI - I can't create one of these objects".
https://github.com/open-telemetry/opentelemetry-dotnet/blob/c7c7a69788f8e70122e2589a8b68baec73c0c815/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OtlpExporterBuilderOptions.cs#L57-L63

Here this is safe to suppress because we are creating instances of those objects in the ctor (which is then created through the DelegatingOptionsFactory:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/c7c7a69788f8e70122e2589a8b68baec73c0c815/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OtlpExporterBuilder.cs#L183-L198

The way the source generated code works is this:

```C#
            if (AsConfigWithChildren(configuration.GetSection("DefaultOptions")) is IConfigurationSection section8)
            {
                global::OpenTelemetry.Exporter.IOtlpExporterOptions? temp10 = instance.DefaultOptions;
                if (temp10 is not null)
                {
                    BindCore(section8, ref temp10, defaultValueIfNotFound: false, binderOptions);
                }
                else
                {
                    throw new InvalidOperationException("Cannot create instance of type 'OpenTelemetry.Exporter.IOtlpExporterOptions' because it is missing a public instance constructor.");
                }
            }
```

It grabs the property first, and if it is a valid object, it binds the configuration to that object. If the object is `null` and someone set configuration for that object, it throws because an interface isn't instantiable.

Reply:
@eerhardt Is there any way to suppress these closer to the thing causing the issue(s) rather than at the project level? I tried to do this...

```csharp
#pragma warning disable SYSLIB1100, SYSLIB1101
        services!.Configure<OtlpExporterBuilderOptions>(name, configuration!);
#pragma warning restore SYSLIB1100, SYSLIB1101
```

...but it doesn't seem to work.

Reply:
According to https://github.com/dotnet/runtime/issues/88865 this is supposed to work, but doesn't for some reason.

I've opened https://github.com/dotnet/runtime/issues/100785 for this issue.